docs/contributing/core/index.md

```diff
@@ -0,0 +1,312 @@
+
+```{important}
+You must {ref}`contributing-sign-and-return-the-plone-contributor-agreement-label` before your contribution can be accepted.
+```
+
+
+## Version support policy
+
+Before you contribute to Plone core, check the [version support policy](https://plone.org/download/release-schedule) to see which versions of Plone are currently supported.
+
+
+## Pre-requisites
+
+It is beyond the scope of this documentation to provide installation instructions for all pre-requisites for your operating system.
+However, the following links and sections below may be helpful.
+
+```{include} ../../volto/contributing/install-operating-system.md
+```
+
+-   Python {SUPPORTED_PYTHON_VERSIONS}
+-   {term}`GNU make`
+-   {term}`Git`
+-   A C compiler
+
+
+### Python
+
+Installing Python is beyond the scope of this documentation.
+However, it is recommended to use a Python version manager, {term}`pyenv` that allows you to install multiple versions of Python on your development environment without destroying your system's Python.
+Plone requires Python version {SUPPORTED_PYTHON_VERSIONS}.
+
+
+### Make
+
+```{include} ../../volto/contributing/install-make.md
+```
+
+
+### Git
+
+```{include} ../../volto/contributing/install-git.md
+```
+
+### C compiler
+
+You need a C compiler on your system to compile some of the Python libraries that Plone uses.
+
+On macOS, Developer Tools provides Clang for a C compiler.
+
+On Linux, [GNU Compiler Collection (GCC)](https://gcc.gnu.org/) is a common option.
+
+
+
+## Install Plone core for development
+
+The tool that installs Plone core is `buildout.coredev`.
+
+The current default and development branch of `buildout.coredev` is `6.1`
+Older versions are named according to their `major.minor` version.
+Its versions align with Plone's `major.minor` versions.
+
+Use a separate directory for each version of Plone to which you want to contribute.
+This will avoid switching between git branches, then re-running buildout, which can cause dependency conflicts between versions of Plone.
+
+To set up a Plone 6 development environment, change your working directory to wherever you place your projects, and clone https://github.com/plone/buildout.coredev.
+You can specify the branch that you want to check out with the `-b` option.
+
+```shell
+cd [MY_PROJECTS]
+# clone a specific major.minor version branch
+git clone -b 6.1 https://github.com/plone/buildout.coredev
+cd buildout.coredev
+```
+
+````{important}
+If you want to use a Python version that is not 3.11, follow these instructions.
+
+Open the file {file}`bootstrap.sh` at the root of the repository.
+Notice that the script expects Python 3.11 to be installed on your system and in your user's `PATH`.
+
+```shell
+#/bin/sh
+`which python3.11` -m venv .
+```
+
+Edit it according to the Python version you want to use, then save and close the file.
```

Comment:
A warning would be good, but it would be even better to allow the developer to use whatever version of Python they want as an argument in the script. I'm not sure how to do that with shell scripts.

Same with the `.bootstrap-pyenv.sh` script.